### ISSUE TITLE
GDExtension export unrealistically requires every feature flag to be part of the library name

### ISSUE THREAD

**Issue by nathanfranke** [godotengine/godot#63227]
### Godot version

v4.0.alpha.custom_build [7378dc6c9]

### System information

Arch on 5.18.11-arch1-1

### Issue description

Exporting my test project to Linux gives the error:
```
Couldn't export extension: res://addons/example/example.gdextension. No suitable library found for export flags: pc, s3tc, linux, 64, debug
```

[According to the code](https://github.com/godotengine/godot/blob/master/editor/plugins/gdextension_export_plugin.h#L60-L110), _all_ feature flags need to be part of the library tags. This means I need to name the library `linux.debug.pc.s3tc.x86_64`.

In addition, the export flags mismatch the GDExtension names (e.g. `64` instead of `x86_64`, `arm64-v8a` instead of `arm64`). This should probably be fixed in https://github.com/godotengine/godot/pull/55778, CC @aaronfranke 

### Steps to reproduce

1) Try to export a project with a GDExtension, you will probably get the error out of the box.

### Minimal reproduction project

https://github.com/nathanfranke/gdextension

**Comment by akien-mga** [comment]
> [According to the code](https://github.com/godotengine/godot/blob/master/editor/plugins/gdextension_export_plugin.h?rgh-link-date=2022-07-20T07%3A57%3A11Z#L60-L110), _all_ feature flags need to be part of the library tags. This means I need to name the library `linux.debug.pc.s3tc.x86_64`. This is unrealistic, as I shouldn't need to update my extension just because I changed the way textures are imported or whatever.

I don't think that's a correct interpretation.

As I read it, this code checks that all tags present in your library name match tags of the platform you export to. Not the other way around. So it can be *any combination* of those flags: `pc`, `s3tc`, `linux`, `64`, `debug`.

E.g. `linux.debug.x86_64`

The more you include, the more specific you make it (so if you do want different libraries for `s3tc` and `etc2` on Android for example, you could). Or if you want to use the same library for `release` and `debug`,  you just use `linux.x86_64`.

**Comment by akien-mga** [comment]
> In addition, the export flags mismatch the GDExtension names (e.g. `64` instead of `x86_64`, `arm64-v8a` instead of `arm64`). This should probably be fixed in #55778, CC @aaronfranke

I believe _this_ is your problem. The feature flag is `64` so that's what you should use probably.

**Comment by nathanfranke** [comment]
Ah, that makes a lot more sense. Thanks for clarifying. Do you think there is any discussion needed still?

**Comment by akien-mga** [comment]
Yeah I think the error message is a bit misleading, could be tweaked further.

**Comment by nathanfranke** [comment]
We can add the junior job label for this one. This error message needs to be more clear: https://github.com/godotengine/godot/blob/master/editor/plugins/gdextension_export_plugin.h#L109

My best error after 15 minutes:
`No suitable library found. Each libraries' tags referred to an invalid feature flag. Feature flags: %s`
